File: leetcode/0429. N-ary Tree Level Order Traversal/Solution.java
// https://leetcode.com/problems/n-ary-tree-level-order-traversal/

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> result = new ArrayList<>();

            Queue<Node> queue = new LinkedList<>();
            queue.add(root);

            while (queue.peek() != null) {
                List<Integer> list = new ArrayList<>();
                Queue<Node> queue2 = new LinkedList<>();

                while (queue.peek() != null) {
                    Node node = queue.poll();

                    list.add(node.val);

                    for (Node nodeChild : node.children) {
                        queue2.add(nodeChild);
                    }
                }

                queue = queue2;
                result.add(list);
            }

            return result;
    }
}

File: leetcode/0429. N-ary Tree Level Order Traversal/Solution.py
# https://leetcode.com/problems/n-ary-tree-level-order-traversal/

"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

from typing import List

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        r = []
        if not root:
            return r

        q = collections.deque()
        q.append(root)

        while q:
            l = []
            for _ in range(len(q)):
                node = q.popleft()
                l.append(node.val)
                q.extend(node.children)

            r.append(l)

        return r